### PR TITLE
Prepend Path to Redis Key

### PR DESCRIPTION
**Pull request by ndellicarpini**
A little weird, but just to give you an idea of what I'm thinking. I attach the path as a `source` to the reference json output by fsspec, then read that source when pulling reading/writing a new key to redis.

### CODE REVIEW

File: redis_fsspec_cache/asyncio.py [hunk 1]
```diff
@@ -97,10 +99,13 @@ def _strip_protocol(path):
     async def _cat_file(self, path, start=None, end=None, **kwargs):
         cached = await self._get_cached(path, start, end)
         if cached is not None:
-            return cached
+            return json.loads(cached)
         chunk = await self.fs._cat_file(path, start=start, end=end, **kwargs)
-        await self._put_cache(chunk, path, start, end)
-        return chunk
+        chunk = re.sub(r'^\s*\{\s*"version"\s*:', '{"source":"' + path + '","version":', chunk.decode())
```
Comment (mpiannucci) [inline redis_fsspec_cache/asyncio.py:104]: Why is this necessary

Comment (ndellicarpini) [reply redis_fsspec_cache/asyncio.py:104]: this is my way to add the path to the key so that you dont need to manually pass it through to `RedisCachingReferenceFileSystem`

Comment (mpiannucci) [reply redis_fsspec_cache/asyncio.py:104]: I dont think i understand. shouldnt the source be available use for the key? And they path is what is queried?

Comment (ndellicarpini) [reply redis_fsspec_cache/asyncio.py:104]: is it a little overcomplicated? maybe? but it would be kinda cringe to create a refs object using a path, then needing to feed both the path and the refs object into the `RedisCachingReferenceFileSystem`

**Comment by mpiannucci** [comment]
Closing in favor of #5
